SwiftWisdom/Core/StandardLibrary/Array/Array+Utilities.swift

```diff
@@ -54,3 +54,43 @@ extension Array {
         }
     }
 }
+
+extension CollectionType {
+    public var ip_middleValues: [Generator.Element]? {
+        guard count > 0 else { return nil }
+        let needsAverageOfTwo = count.toIntMax().ip_isEven
+        let middle = startIndex.advancedBy(count / 2)
+        if needsAverageOfTwo {
+            let leftOfMiddle = startIndex.advancedBy((count / 2) - 1)
+            return [self[middle], self[leftOfMiddle]]
+        } else {
+            return [self[middle]]
+        }
+    }
+}
+
+extension SequenceType where Generator.Element: Equatable {
+    public func ip_mostCommonElements() -> [Generator.Element]? {
```

Comment:
I think this one could also be more thoroughly defined, some of the behavior will be unexpected.  ie: `[1,2,3,4,5]` would return the first element. I think some of these should be moved to another PR so we can get the sure things merged in